string[] order_list
int idx
